web/src/lib/search/searchConfig.ts

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { makeSnippet } from "@/lib/search/helpers";
+import { ConfigMatch, GroupedConfigMatch } from "@/lib/search/typesForSearch";
+
+export const searchConfig = (obj: any, term: string, cmsConfig: any): GroupedConfigMatch[] => {
+  const configMatches = searchObj(obj.default, term, [], []).map((it) => {
+    const cmsPath = findCmsConfigPath(cmsConfig, it.keyPath);
+    return {
+      value: makeSnippet(it.value, term),
+      cmsLabelPath: cmsPath,
+    };
+  });
+
+  return groupByCMSFile(configMatches);
+};
+
+const groupByCMSFile = (configMatches: ConfigMatch[]): GroupedConfigMatch[] => {
+  const groupedConfigMatches: Record<string, ConfigMatch[]> = {};
+
+  for (const match of configMatches) {
+    const groupLabel = match.cmsLabelPath.slice(0, 2).join(" > ");
+    const existingGroup = groupedConfigMatches[groupLabel];
+    if (existingGroup) {
+      groupedConfigMatches[groupLabel] = [...existingGroup, match];
+    } else {
+      groupedConfigMatches[groupLabel] = [match];
+    }
+  }
+
+  return Object.keys(groupedConfigMatches).map((key) => ({
+    groupLabelPath: key,
+    matches: groupedConfigMatches[key],
+  }));
+};
+
+const searchObj = (obj: any, term: string, matches: JsonMatch[], keyPaths: string[]): JsonMatch[] => {
+  if (typeof obj === "object" && obj !== null && !Array.isArray(obj)) {
+    for (const key of Object.keys(obj)) {
+      const value = obj[key];
+      if (typeof value === "string") {
+        if (value.toLowerCase().includes(term)) {
+          matches = [
+            ...matches,
+            {
+              value: value,
+              keyPath: [...keyPaths, key],
+            },
+          ];
+        }
+      } else if (typeof value === "object" && value !== null && !Array.isArray(value)) {
```

Comment:
This needs to be validated, but can this just be an `else`? Because we're doing the same check on line 38 with a similar result I feel like this could be simplified.